tests/cypress/e2e/api/graphqlPermissions.cy.ts

```diff
@@ -0,0 +1,182 @@
+    addUserToGroup
+} from '@jahia/cypress';
+
+describe('Test graphql permissions', () => {
+    let subList2Cursor: string;
+
+    before('create user and test data', () => {
+        // DeleteNode('/testList'); // TODO should not be needed
+        createUser('testUser', 'testPassword');
+        addUserToGroup('testUser', 'privileged');
+
+        /*
+        Test data to create:
+        /testList (jnt:contentList) - owner: true
+            /testSubList1 (jnt:contentList) - owner: false
+            /testSubList2 (jnt:contentList) - owner: true
+            /reference1 (jnt:contentReference) - owner: false
+                [j:node=ref->/testSubList2]
+            /reference2 (jnt:contentReference) - owner: true
+                [j:node=ref->/testSubList2]
+         */
+
+        addNode({
+            parentPathOrId: '/',
+            name: 'testList',
+            primaryNodeType: 'jnt:contentList',
+            children: [
+                {
+                    name: 'testSubList1',
+                    primaryNodeType: 'jnt:contentList'
+                },
+                {
+                    name: 'testSubList2',
+                    primaryNodeType: 'jnt:contentList'
+                },
+                {
+                    name: 'reference1',
+                    primaryNodeType: 'jnt:contentReference'
+                },
+                {
+                    name: 'reference2',
+                    primaryNodeType: 'jnt:contentReference'
+                }
+            ]
+        })
+            .then(response => {
+                // Keep reference to testSubList2
+                subList2Cursor = response.data.jcr.addNode.addChildrenBatch[1].uuid;
+                setNodeProperty('/testList/reference1', 'j:node', subList2Cursor, 'en');
+                setNodeProperty('/testList/reference2', 'j:node', subList2Cursor, 'en');
+            })
+            .then(() => {
+                // Tweak the permissions
+                grantRoles('/testList', ['owner'], 'testUser', 'USER');
+                grantRoles('/testList', ['privileged'], 'testUser', 'USER');
+                revokeRoles('/testList/testSubList1', ['owner'], 'testUser', 'USER');
+                revokeRoles('/testList/reference1', ['owner'], 'testUser', 'USER');
+            }
+            );
+    });
+
+    after('Delete user and test data', () => {
+        deleteUser('testUser');
+        cy.apolloClient() // TODO needed but I don't know why
+            .apollo({
+                mutationFile: 'jcr/deleteNode.graphql',
+                variables: {
+                    pathOrId: '/testList'
+                }
+            });
+    });
+
+    it('Should get error not retrieve protected node', () => {
+        cy.apolloClient({username: 'testUser', password: 'testPassword'})
+            .apollo({
+                query: gql`
+                query {
+                    currentUser {
+                        username
+                    }
```

Comment:
If you don't test anything on the current user maybe it can be removed from the query to avoid confusion and focus on the test. 
Anyway you can also add assertion on the username..

Reply:
you're right, adding the `currentUser` here is useless, removing it